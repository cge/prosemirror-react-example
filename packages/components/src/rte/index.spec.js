import Rte from '.';
import renderer from 'react-test-renderer';

describe('Rte', () => {
  it('renders without crashing', () => {
    shallow(<Rte id="hello" onChange={() => {}} />);
  });

  it('renders correctly', () => {
    let tree = renderer.create(<Rte id="hello" onChange={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
