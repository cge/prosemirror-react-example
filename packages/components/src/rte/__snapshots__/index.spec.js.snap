// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rte renders correctly 1`] = `
<div
  className="rte"
>
  <div
    className="rte-toolbar"
  >
    <div
      className="rte-toolbar-segment-container"
    >
      <div
        className="rte-toolbar-segment"
      >
        <div
          className="rte-toolbar-button"
          onClick={[Function]}
        >
          <div
            className="rte-toolbar-button-icon"
            style={
              Object {
                "fontSize": 16,
                "opacity": 0.3,
              }
            }
          >
            <span
              aria-label="undo"
              className="anticon anticon-undo"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="undo"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M511.4 124C290.5 124.3 112 303 112 523.9c0 128 60.2 242 153.8 315.2l-37.5 48c-4.1 5.3-.3 13 6.3 12.9l167-.8c5.2 0 9-4.9 7.7-9.9L369.8 727a8 8 0 00-14.1-3L315 776.1c-10.2-8-20-16.7-29.3-26a318.64 318.64 0 01-68.6-101.7C200.4 609 192 567.1 192 523.9s8.4-85.1 25.1-124.5c16.1-38.1 39.2-72.3 68.6-101.7 29.4-29.4 63.6-52.5 101.7-68.6C426.9 212.4 468.8 204 512 204s85.1 8.4 124.5 25.1c38.1 16.1 72.3 39.2 101.7 68.6 29.4 29.4 52.5 63.6 68.6 101.7 16.7 39.4 25.1 81.3 25.1 124.5s-8.4 85.1-25.1 124.5a318.64 318.64 0 01-68.6 101.7c-7.5 7.5-15.3 14.5-23.4 21.2a7.93 7.93 0 00-1.2 11.1l39.4 50.5c2.8 3.5 7.9 4.1 11.4 1.3C854.5 760.8 912 649.1 912 523.9c0-221.1-179.4-400.2-400.6-399.9z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="rte-toolbar-button"
          onClick={[Function]}
        >
          <div
            className="rte-toolbar-button-icon"
            style={
              Object {
                "fontSize": 16,
                "opacity": 0.3,
              }
            }
          >
            <span
              aria-label="redo"
              className="anticon anticon-redo"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="redo"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M758.2 839.1C851.8 765.9 912 651.9 912 523.9 912 303 733.5 124.3 512.6 124 291.4 123.7 112 302.8 112 523.9c0 125.2 57.5 236.9 147.6 310.2 3.5 2.8 8.6 2.2 11.4-1.3l39.4-50.5c2.7-3.4 2.1-8.3-1.2-11.1-8.1-6.6-15.9-13.7-23.4-21.2a318.64 318.64 0 01-68.6-101.7C200.4 609 192 567.1 192 523.9s8.4-85.1 25.1-124.5c16.1-38.1 39.2-72.3 68.6-101.7 29.4-29.4 63.6-52.5 101.7-68.6C426.9 212.4 468.8 204 512 204s85.1 8.4 124.5 25.1c38.1 16.1 72.3 39.2 101.7 68.6 29.4 29.4 52.5 63.6 68.6 101.7 16.7 39.4 25.1 81.3 25.1 124.5s-8.4 85.1-25.1 124.5a318.64 318.64 0 01-68.6 101.7c-9.3 9.3-19.1 18-29.3 26L668.2 724a8 8 0 00-14.1 3l-39.6 162.2c-1.2 5 2.6 9.9 7.7 9.9l167 .8c6.7 0 10.5-7.7 6.3-12.9l-37.3-47.9z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        className="rte-toolbar-segment"
      >
        <div
          className="rte-toolbar-button"
          onClick={[Function]}
        >
          <div
            className="rte-toolbar-button-icon"
            style={
              Object {
                "fontSize": 16,
                "opacity": 0.3,
              }
            }
          >
            <span
              aria-label="bold"
              className="anticon anticon-bold"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="bold"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M697.8 481.4c33.6-35 54.2-82.3 54.2-134.3v-10.2C752 229.3 663.9 142 555.3 142H259.4c-15.1 0-27.4 12.3-27.4 27.4v679.1c0 16.3 13.2 29.5 29.5 29.5h318.7c117 0 211.8-94.2 211.8-210.5v-11c0-73-37.4-137.3-94.2-175.1zM328 238h224.7c57.1 0 103.3 44.4 103.3 99.3v9.5c0 54.8-46.3 99.3-103.3 99.3H328V238zm366.6 429.4c0 62.9-51.7 113.9-115.5 113.9H328V542.7h251.1c63.8 0 115.5 51 115.5 113.9v10.8z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        className="rte-toolbar-segment"
      >
        <div
          className="rte-toolbar-button"
          onClick={[Function]}
        >
          <div
            className="rte-toolbar-button-icon"
            style={
              Object {
                "fontSize": 16,
                "opacity": 1,
              }
            }
          >
            <span
              aria-label="ordered-list"
              className="anticon anticon-ordered-list"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="ordered-list"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M920 760H336c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0-568H336c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H336c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM216 712H100c-2.2 0-4 1.8-4 4v34c0 2.2 1.8 4 4 4h72.4v20.5h-35.7c-2.2 0-4 1.8-4 4v34c0 2.2 1.8 4 4 4h35.7V838H100c-2.2 0-4 1.8-4 4v34c0 2.2 1.8 4 4 4h116c2.2 0 4-1.8 4-4V716c0-2.2-1.8-4-4-4zM100 188h38v120c0 2.2 1.8 4 4 4h40c2.2 0 4-1.8 4-4V152c0-4.4-3.6-8-8-8h-78c-2.2 0-4 1.8-4 4v36c0 2.2 1.8 4 4 4zm116 240H100c-2.2 0-4 1.8-4 4v36c0 2.2 1.8 4 4 4h68.4l-70.3 77.7a8.3 8.3 0 00-2.1 5.4V592c0 2.2 1.8 4 4 4h116c2.2 0 4-1.8 4-4v-36c0-2.2-1.8-4-4-4h-68.4l70.3-77.7a8.3 8.3 0 002.1-5.4V432c0-2.2-1.8-4-4-4z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="rte-toolbar-button"
          onClick={[Function]}
        >
          <div
            className="rte-toolbar-button-icon"
            style={
              Object {
                "fontSize": 16,
                "opacity": 1,
              }
            }
          >
            <span
              aria-label="unordered-list"
              className="anticon anticon-unordered-list"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="unordered-list"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M912 192H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 284H328c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h584c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM104 228a56 56 0 10112 0 56 56 0 10-112 0zm0 284a56 56 0 10112 0 56 56 0 10-112 0zm0 284a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="rte-toolbar-separator"
        />
        <div
          className="rte-toolbar-button"
          onClick={[Function]}
        >
          <div
            className="rte-toolbar-button-icon"
            style={
              Object {
                "fontSize": 16,
                "opacity": 0.3,
              }
            }
          >
            <span
              aria-label="menu-fold"
              className="anticon anticon-menu-fold"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="menu-fold"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="rte-content"
    onClick={[Function]}
    style={
      Object {
        "height": 160,
        "minHeight": 160,
      }
    }
  >
    <div
      className="rte-editor"
    />
  </div>
  <div
    className="rte-footer"
  >
    <div
      className="rte-footer-resizer"
      onMouseDown={[Function]}
    >
      â—¢
    </div>
  </div>
</div>
`;
